src/base/bittorrent/session.h

```diff
@@ -237,6 +237,14 @@ namespace BitTorrent
         virtual Path finishedTorrentExportDirectory() const = 0;
         virtual void setFinishedTorrentExportDirectory(const Path &path) = 0;
 
+        virtual bool isAddTrackersFromURLEnabled() const = 0;
+        virtual void setAddTrackersFromURLEnabled(bool enabled) = 0;
+        virtual QString additionalTrackersURL() const = 0;
+        virtual void setAdditionalTrackersURL(const QString &url) = 0;
+        virtual QString additionalTrackersFromURL() const = 0;
+        virtual void setAdditionalTrackersFromURL(const QString &trackers) = 0;
```

Comment:
It looks very confusing. Why should it be editable if it is supposed to be loaded from external source?

Reply:
This function is how we persist the trackers list loaded from the URL. Though I can change this to be private.